src/controllers/controller.doughnut.js

```diff
@@ -76,7 +76,12 @@ module.exports = function(Chart) {
 
 				for (i = 0, ilen = (chart.data.datasets || []).length; i < ilen; ++i) {
 					meta = chart.getDatasetMeta(i);
-					meta.data[index].hidden = !meta.data[index].hidden;
+					try {
```

Comment:
Is it only to avoid accessing an undefined index of data? If so, wouldn't this be more efficient and easier to understand than using exceptions?

```javascript
if (meta && meta.data[index]) {
    meta.data[index].hidden = !meta.data[index].hidden;
}
```